twake/frontend/src/app/scenes/Client/ChannelsBar/ChannelTemplateEditor.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { FC, useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import Languages from 'services/languages/languages.js';
+import InputWithIcon from 'components/Inputs/InputWithIcon.js';
+import AutoHeight from 'components/AutoHeight/AutoHeight.js';
+import Select from 'components/Select/Select.js';
+import { ChannelType } from 'app/models/Channel';
+import {
+  ObjectModalSeparator,
+  ObjectModalSectionTitle,
+} from 'components/ObjectModal/ObjectModal.js';
+import RouterServices from 'services/RouterServices';
+import OldCollections from 'services/Depreciated/Collections/Collections';
+
+type Props = {
+  channel: ChannelType | undefined;
+  onChange: (channelEntries: any) => void;
+};
+
+const ChannelTemplateEditor: FC<Props> = ({ channel, onChange }) => {
+  const [icon, setIcon] = useState<string>(channel?.icon || '');
+  const [name, setName] = useState<string>(channel?.name || '');
+  const [description, setDescription] = useState<string>(channel?.description || '');
+  const [visibility, setVisibility] = useState<string>(channel?.visibility || 'private');
+
+  // TODO find a better way to get workspace id
+  const params: any = useParams();
+  const workspaceId = RouterServices.translateToUUID(params.workspaceId);
+
+  useEffect(() => {
+    onChange({
+      icon,
+      name,
+      description,
+      visibility,
+      // TODO find a better way to get company id
+      company_id: OldCollections.get('workspaces').find(workspaceId)?.group?.id,
+      workspace_id: workspaceId,
```

Comment:
I don't want this to come from the template editor, please move this code to twake/frontend/src/app/scenes/Client/ChannelsBar/ChannelWorkspaceEditor.tsx

Reply:
done